force-app/main/default/classes/DMLOperations.cls

```diff
@@ -179,41 +227,67 @@ public with sharing class DMLOperations {
      * Example Contact.LastName = 'Jane' Link to Account.Name = 'Jane'
 	 * @param contactsToUpsert List of Contacts to be upserted and linked to accounts.
 	 */
-	public static void upsertAccountsWithContacts(List<Contact> contactsToUpsert) {
-		// Loop through each Contact
+            public static void upsertAccountsWithContacts(List<Contact> contactsToUpsert) {
+                
+                List<Contact> contacts = new List<Contact>();
```

Comment:
Here you don't actually need to create a new `List<Contact> contacts`. You can simply `upsert` the list passed into the method: `contactsToUpsert`. This is because you're actually editing the Contacts in that original list. See below.